content/client-lib-development-guide/features.textile

```diff
@@ -444,17 +453,28 @@ h3(#realtime-channel). Channel
 * @(RTP4)@ Ensure a test exists that enters 250 members using @Presence#enterClient@ on a single connection, and checks for @PRESENT@ events to be emitted on another connection for each member, and once sync is complete, all 250 members should be present in a @Presence#get@ request
 * @(RTP5)@ Channel state change side effects:
-** @(RTP5a)@ If the channel enters the @DETACHED@ or @FAILED@ state then all queued presence messages will fail immediately, and the presence map is cleared
-** @(RTP5b)@ If a channel enters the @ATTACHED@ state then all queued presence messages will be sent immediately and a presence @SYNC@ will be initiated implicitly
+** @(RTP5a)@ If the channel enters the @DETACHED@ or @FAILED@ state then all queued presence messages will fail immediately, and the @PresenceMap@ and "internal PresenceMap (see RTP17)":#RTP7 is cleared. The latter ensures members are not automatically re-entered if the @Channel@ later becomes attached
+** @(RTP5f)@ If the channel enters the @SUSPENDED@ state then all queued presence messages will fail immediately, and the @PresenceMap@ is cleared
+** @(RTP5b)@ If a channel enters the @ATTACHED@ state then all queued presence messages will be sent immediately and a presence @SYNC@ may be initiated by the Ably service or the Ably service may indicate there is no @SYNC@ necessary
+** @(RTP5c)@ In addition, when a channel becomes @ATTACHED@, the following must happen in regards to @Presence@:
+*** @(RTP5c1)@ If the "@resumed@ flag":#TH4 is true and no @SYNC@ is initiated as part of the attach, do nothing as there is no loss of continuity on the channel and no change to presence. The @PresenceMap@ is not effected and no members need to be re-entered
+*** @(RTP5c2)@ If the "@resumed@ flag":#TH4 is true and a @SYNC@ is initiated as part of the attach, then once the @SYNC@ is complete, all members not present in the
+@PresenceMap@ but present in the "internal @PresenceMap@":#RTP17 must be re-entered automatically by the client using the @clientId@ and @data@ attributes from each. The members re-entered automatically must be removed from the "internal @PresenceMap@":#RTP17 ensuring that members present on the channel are constructed from presence events sent from Ably since the channel became @ATTACHED@
+*** @(RTP5c3)@ If the "@resumed@ flag":#TH4 is false, then the @PresenceMap@ should be cleared, and once the @SYNC@ is complete, all members in the "internal @PresenceMap@":#RTP17 must be re-entered automatically by the client using the @clientId@ and @data@ attributes from each. The "internal @PresenceMap@":#RTP17 must then be cleared to ensure that members present on the channel are constructed from presence events sent from Ably since the channel became @ATTACHED@
```

Comment:
""internal @PresenceMap@":#RTP17 must then be cleared to ensure that members present on the channel are constructed from presence events sent from Ably since the channel became @ATTACHED@" -> I don't see why it's not handled just like any other sync - you don't drain the internal map completely, but remove the members that are outstanding on completion of the sync.